test/github-package.test.js

```diff
@@ -8,617 +8,836 @@ import {fileExists, getTempDir} from '../lib/helpers';
 import GithubPackage from '../lib/github-package';
 
 describe('GithubPackage', function() {
-  let atomEnv, workspace, project, commands, notificationManager, grammars, config, keymaps;
-  let confirm, tooltips, styles;
-  let getLoadSettings, currentWindow, configDirPath, deserializers;
-  let githubPackage, contextPool;
-
-  beforeEach(async function() {
-    atomEnv = global.buildAtomEnvironment();
-    await disableFilesystemWatchers(atomEnv);
-
-    workspace = atomEnv.workspace;
-    project = atomEnv.project;
-    commands = atomEnv.commands;
-    deserializers = atomEnv.deserializers;
-    notificationManager = atomEnv.notifications;
-    tooltips = atomEnv.tooltips;
-    config = atomEnv.config;
-    keymaps = atomEnv.keymaps;
-    confirm = atomEnv.confirm.bind(atomEnv);
-    styles = atomEnv.styles;
-    grammars = atomEnv.grammars;
-    getLoadSettings = atomEnv.getLoadSettings.bind(atomEnv);
-    currentWindow = atomEnv.getCurrentWindow();
-    configDirPath = path.join(__dirname, 'fixtures', 'atomenv-config');
-
-    githubPackage = new GithubPackage({
-      workspace, project, commands, notificationManager, tooltips, styles, grammars,
-      keymaps, config, deserializers,
-      confirm, getLoadSettings, currentWindow,
-      configDirPath,
-      renderFn: sinon.stub().callsFake((component, element, callback) => {
-        if (callback) {
-          process.nextTick(callback);
-        }
-      }),
-    });
-
-    contextPool = githubPackage.getContextPool();
-  });
-
-  afterEach(async function() {
-    await githubPackage.deactivate();
-
-    atomEnv.destroy();
-  });
-
-  async function contextUpdateAfter(chunk) {
+  async function contextUpdateAfter(githubPackage, chunk) {
     const updatePromise = githubPackage.getSwitchboard().getFinishActiveContextUpdatePromise();
     await chunk();
     return updatePromise;
   }
 
   describe('construction', function() {
-    let githubPackage1;
+    let atomEnv;
+    beforeEach(async function() {
+      atomEnv = global.buildAtomEnvironment();
+      await disableFilesystemWatchers(atomEnv);
+    });
 
-    afterEach(async function() {
-      if (githubPackage1) {
-        await githubPackage1.deactivate();
-      }
+    afterEach(function() {
+      atomEnv.destroy();
     });
 
     async function constructWith(projectPaths, initialPaths) {
       const realProjectPaths = await Promise.all(
         projectPaths.map(projectPath => getTempDir({prefix: projectPath})),
       );
 
+      const {
+        workspace, project, commands, notificationManager, tooltips,
+        deserializers, config, keymaps, styles, grammars,
+      } = atomEnv;
+
+      const confirm = atomEnv.confirm.bind(atomEnv);
+      const currentWindow = atomEnv.getCurrentWindow();
+      const configDirPath = path.join(__dirname, 'fixtures', 'atomenv-config');
```

Comment:
Oh, fun fact: `path.join()` implicitly normalizes path separators. So you can write this a tad more compactly:

```suggestion
      const configDirPath = path.join(__dirname, 'fixtures/atomenv-config');
```

And it works on Windows:

```
> const path = require('path')
undefined
> path.join('a/b', 'c/d/e')
'a\\b\\c\\d\\e'
>
```

Not a big deal here, but can sometimes save you a lot of `"", "", ""` stuff.